src/components/SearchBar.js

```diff
@@ -3,17 +3,27 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { withRouter } from 'react-router-dom';
 
-import Dropdown from 'react-bootstrap/lib/Dropdown';
-import Glyphicon from 'react-bootstrap/lib/Glyphicon';
-import MenuItem from 'react-bootstrap/lib/MenuItem';
+import {
+  Dropdown,
+  Glyphicon,
+  MenuItem,
+  Button,
+  Modal,
+  Row,
+  Col,
+} from 'react-bootstrap';
 
 import Configurable from './Configurable';
 import AutoCompleteInput from './AutoCompleteInput';
+import AuthUtils from '../util/AuthUtils';
+import SimpleQueryRequest from '../api/SimpleQueryRequest';
+import QueryResponse from '../api/QueryResponse';
 
 declare var webkitSpeechRecognition: any; // Prevent complaints about this not existing
 
 type SearchBarProps = {
   history: PropTypes.object.isRequired;
+  location: PropTypes.object.isRequired;
```

Comment:
If we make the `location` required this would become a breaking change. Can you refactor things a bit to make this optional?